packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2196,6 +2196,14 @@ export function captureCommitPhaseError(
   nearestMountedAncestor: Fiber | null,
   error: mixed,
 ) {
+  if (__DEV__) {
+    // Commit phase doesn't use invokeGuardedCallback anymore due to perf.
+    // However, we still want to print the message and make it fire onerror.
+    invokeGuardedCallback(null, () => {
```

Comment:
Note I could move it further down, e.g. only if it's caught by error boundary. Then boundary-less apps don't double-log the error in effects. However, they'd still double-log rendering errors. So I don't think this special case is worth it compared to having to remember why it's there and what it does.